### ISSUE TITLE
Postgres throws error everytime

### ISSUE THREAD

**Issue by worldspawn** [chucknorris/roundhouse#206]
After a successful execution roundhouse errors with 

```
Cannot access a disposed object.
Object name: 'NpgsqlConnection'.
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'NpgsqlConnection'.
   at Npgsql.NpgsqlConnection.CheckNotDisposed()
   at Npgsql.NpgsqlConnection.get_FullState()
   at Npgsql.NpgsqlConnection.get_State()
   at roundhouse.databases.DefaultDatabase`1.dispose_connection(IConnection`1 connection)
   at roundhouse.databases.DefaultDatabase`1.Dispose()
   at roundhouse.runners.RoundhouseMigrationRunner.run()
   at roundhouse.console.Program.run_migrator(ConfigurationPropertyHolder configuration)
   at roundhouse.console.Program.Main(String[] args)
```

Its pretty shit that .State throws an exception but it does.

**Comment by lukemcgregor** [comment]
+1 i have this also

**Comment by worldspawn** [comment]
Theres an issue on the npgsql side complaining that State throws when disposed when it should just return Closed.

The issue happens for postgre for the above reason but the migration runner closes the connection and then closes it again in a finally block.

See: 
- https://github.com/chucknorris/roundhouse/blob/1548be8c8287b0618f3214b30e787197bf2255e6/product/roundhouse/runners/RoundhouseMigrationRunner.cs#L176
- https://github.com/chucknorris/roundhouse/blob/1548be8c8287b0618f3214b30e787197bf2255e6/product/roundhouse/runners/RoundhouseMigrationRunner.cs#L205

This 176 calls close_connection which disposes the connection. A simple fix might be to set a bool in the runner to true immediately after calling close_connection then then doing `if (!conn_disposed)` in the finally.

**Comment by worldspawn** [comment]
https://github.com/chucknorris/roundhouse/pull/208

**Comment by lukemcgregor** [comment]
Its fortunate the actual migration is successful still

**Comment by ChrisMissal** [comment]
Closed by #208
